provider/server/machine_create_test.go

```diff
@@ -24,13 +28,86 @@ var _ = Describe("CreateMachine", func() {
 				Spec: &iri.MachineSpec{
 					Power: iri.Power_POWER_ON,
 					Image: &iri.ImageSpec{
-						Image: "example.org/foo:latest",
+						Image: imageURL,
 					},
-					Class: machineClassx3xlarge,
+					Class:        machineClassx3xlarge,
+					IgnitionData: ignitionData,
+					// TODO: will be done when convertMachineToIRIMachine() supports Volumes and NetworkInterfaces
+					Volumes:           nil,
+					NetworkInterfaces: nil,
 				},
 			},
 		})
 		Expect(err).NotTo(HaveOccurred())
-		Expect(res).NotTo(BeNil())
+
+		By("ensuring the correct creation response")
+		Expect(createResp).Should(SatisfyAll(
+			HaveField("Machine.Metadata.Id", Not(BeEmpty())),
+			HaveField("Machine.Spec.Power", iri.Power_POWER_ON),
+			HaveField("Machine.Spec.Image.Image", Equal(imageURL)),
+			HaveField("Machine.Spec.Class", machineClassx3xlarge),
+			HaveField("Machine.Spec.IgnitionData", Equal(ignitionData)),
+			HaveField("Machine.Spec.Volumes", BeNil()),
+			HaveField("Machine.Spec.NetworkInterfaces", BeNil()),
+			HaveField("Machine.Status.ObservedGeneration", BeZero()),
+			HaveField("Machine.Status.State", Equal(iri.MachineState_MACHINE_PENDING)),
+			HaveField("Machine.Status.ImageRef", BeEmpty()),
+			HaveField("Machine.Status.Volumes", BeNil()),
+			HaveField("Machine.Status.NetworkInterfaces", BeNil()),
+		))
+
+		DeferCleanup(func(ctx SpecContext) {
+			_, err := machineClient.DeleteMachine(ctx, &iri.DeleteMachineRequest{MachineId: createResp.Machine.Metadata.Id})
+			Expect(err).ShouldNot(HaveOccurred())
+			Eventually(func() bool {
+				_, err = libvirtConn.DomainLookupByUUID(libvirtutils.UUIDStringToBytes(createResp.Machine.Metadata.Id))
+				return libvirt.IsNotFound(err)
+			}).Should(BeTrue())
+		})
+
+		By("ensuring domain and domain XML is created for machine")
+		var domain libvirt.Domain
+		Eventually(func() error {
+			domain, err = libvirtConn.DomainLookupByUUID(libvirtutils.UUIDStringToBytes(createResp.Machine.Metadata.Id))
+			return err
+		}).Should(Succeed())
+		domainXMLData, err := libvirtConn.DomainGetXMLDesc(domain, 0)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(domainXMLData).NotTo(BeEmpty())
+
+		By("ensuring domain for machine is in running state")
+		Eventually(func() libvirt.DomainState {
+			domainState, _, err := libvirtConn.DomainGetState(domain, 0)
+			Expect(err).NotTo(HaveOccurred())
+			return libvirt.DomainState(domainState)
+		}).Should(Equal(libvirt.DomainRunning))
+
+		By("updating machine annotations")
+		Eventually(func() error {
+			_, err := machineClient.UpdateMachineAnnotations(ctx, &iri.UpdateMachineAnnotationsRequest{
```

Comment:
It is not needed in this test, as `UpdateMachineAnnotations` is not defined in machine_create.go. Rather, it can be properly validated in the test of machine_annotations_update.go. 
However, If you still wish to keep it here as well, please validate if the annotation is correctly set. But, that will be just a duplication of work.

Reply:
Currently CreateMachine() always returns `PENDING` state for new machine. For more details please refer this: https://github.com/ironcore-dev/libvirt-provider/issues/102
Once this gets fixed we wont need `UpdateMachineAnnotations` anymore here. As of now its been added as an workaround to get latest domain state and update it in machinestore.

Reply:
I get the machine running state. So, I would like to differ here. As per the design, 1st reconciliation sets the machine state to pending and it gets the running state in the next iteration. 
The controller is working fine as I have observed. 

Let's take this offline.

Reply:
Removed `UpdateMachineAnnotations` workaround and added empty disk volume to create machine successfully. We can have test without any volume when #102 gets fixed.